components/eam/src/physics/crm/cloud_diagnostics.F90

```diff
@@ -1,525 +1,374 @@
+  call pbuf_get_field(pbuf, cld_idx, cld, start=(/1,1,itim_old/), kount=(/pcols,pver,1/) )
+
+  call pbuf_get_field(pbuf, iclwp_idx, iclwp )
+  call pbuf_get_field(pbuf, iciwp_idx, iciwp )
+  call pbuf_get_field(pbuf, dei_idx, dei )
+  call pbuf_get_field(pbuf, mu_idx, mu )
+  call pbuf_get_field(pbuf, lambda_idx, lambda )
+
+  call outfld('dei_cloud',dei(:,:),pcols,lchnk)
+  call outfld('mu_cloud',mu(:,:),pcols,lchnk)
+  call outfld('lambda_cloud',lambda(:,:),pcols,lchnk)
+
+  ! call pbuf_get_field(pbuf, rei_idx, rei )
+  ! call pbuf_get_field(pbuf, rel_idx, rel )
+  ! call outfld('rel_cloud', rel, pcols, lchnk)
+  ! call outfld('rei_cloud', rei, pcols, lchnk)
+
+  call pbuf_get_field(pbuf, cicewp_idx, cicewp )
+  call pbuf_get_field(pbuf, cliqwp_idx, cliqwp )
+
+  ! Compute liquid and ice water paths
+
+  ! ----------------------------------------------------------- !
+  ! Adjust in-cloud water values to take account of convective  !
+  ! in-cloud water. It is used to calculate the values of       !
+  ! iclwp and iciwp to pass to the radiation.                   !
+  ! ----------------------------------------------------------- !
+  if( conv_water_in_rad /= 0 ) then
+    allcld_ice(:ncol,:) = 0._r8 ! Grid-avg all cloud liquid
+    allcld_liq(:ncol,:) = 0._r8 ! Grid-avg all cloud ice
+
+    call conv_water_4rad( state, pbuf, conv_water_in_rad, allcld_liq, allcld_ice )
+  else
+    allcld_liq(:ncol,top_lev:pver) = state%q(:ncol,top_lev:pver,ixcldliq)  ! Grid-ave all cloud liquid
+    allcld_ice(:ncol,top_lev:pver) = state%q(:ncol,top_lev:pver,ixcldice)  !           "        ice
+  end if
+
+  ! ------------------------------------------------------------ !
+  ! Compute in cloud ice and liquid mixing ratios                !
+  ! Note that 'iclwp, iciwp' are used for radiation computation. !
+  ! ------------------------------------------------------------ !
+  do k = 1,pver
+    do i = 1,ncol
+      iciwp(i,k)  = 0._r8
+      iclwp(i,k)  = 0._r8
+      ! icimr(i,k)  = 0._r8
+      ! icwmr(i,k)  = 0._r8
+      ! iwc(i,k)    = 0._r8
+      ! lwc(i,k)    = 0._r8
+      gicewp(i,k) = 0._r8
+      gliqwp(i,k) = 0._r8
+      cicewp(i,k) = 0._r8
+      cliqwp(i,k) = 0._r8
+    end do
+  end do
+
+  ! do k = top_lev, pver
+  !   do i = 1, ncol
+  !     ! Limits for in-cloud mixing ratios consistent with MG microphysics
+  !     ! in-cloud mixing ratio maximum limit of 0.005 kg/kg
+  !     icimr(i,k)     = min( allcld_ice(i,k) / max(0.0001_r8,cld(i,k)),0.005_r8 )
+  !     icwmr(i,k)     = min( allcld_liq(i,k) / max(0.0001_r8,cld(i,k)),0.005_r8 )
+  !     iwc(i,k)       = allcld_ice(i,k) * state%pmid(i,k) / (287.15_r8*state%t(i,k))
+  !     lwc(i,k)       = allcld_liq(i,k) * state%pmid(i,k) / (287.15_r8*state%t(i,k))
+  !     ! Calculate total cloud water paths in each layer
+  !     iciwp(i,k)     = icimr(i,k) * state%pdel(i,k) / gravit
+  !     iclwp(i,k)     = icwmr(i,k) * state%pdel(i,k) / gravit
+  !   end do
+  ! end do
+
+  ! do k = 1,pver
```

Comment:
Is this a mistake? It looks like gliqwp and gicewp never get set, but they are output below.

Reply:
I guess I forgot to go back and clean this up. I think I've got it all sorted so these variables will be calculated like they used to be.

Reply:
ok, so in the old/current version, these calculations were all done here, but then there was additional logic to just not write it to the history files when using the MMF. I think that's why I commented this all out, since I wasn't sure if deleting it would cause a problem elsewhere. Now I feel pretty confident that there's a lot of useless stuff here, but not all of it, so I'm gonna try to delete all this extra/duplicate stuff.

Reply:
Alright, I got this file cleaned up in a recent commit. There aren't any commented lines floating around, but I should do some work to verify that the radiation calculations weren't messed up from deleting what I determined to be useless calculations.